lib/resque/plugins/timed_round_robin/resque_worker.rb

```diff
@@ -18,7 +18,14 @@ def self.working
   class DataStore
     def working_keys(worker_ids)
       redis_keys = worker_ids.map { |id| "worker:#{id}" }
-      redis_keys.select { |k| @redis.exists(k) }
+
+      # pipeline our exists checks since there is no MEXISTS commmand.
+      # more efficient as we don't need a round trip between each query.
+      key_exists = @redis.pipelined { redis_keys.each { |k| @redis.exists(k) } }
+
+      # this returns a [true, false, true] type array.
+      # filter our original list for values where true was returned.
+      redis_keys.zip(key_exists).map { |key, exists| key if exists }.compact
```

Comment:
I'm not a fan of `map.compact`, but that's more of a personal preference.  You could write this using `each_with_object`:
```
redis_keys.zip(key_exists).each_with_object([]){|pair, keys| keys << pair[0] if pair[1] }
```
but I don't really love that.  Another option is to pass a block to `zip`, which forgoes chained methods entirely:
```
keys = []
redis_keys.zip(key_exists) {|key, exists| keys << key if exists}
keys
```